include/oneapi/dpl/pstl/glue_algorithm_ranges_impl.h

```diff
@@ -1099,11 +1099,9 @@ struct __reverse_fn
     {
         const auto __dispatch_tag = oneapi::dpl::__ranges::__select_backend(__exec);
 
-        auto __last = std::ranges::begin(__r) + std::ranges::size(__r);
-
         oneapi::dpl::__internal::__ranges::__pattern_reverse(__dispatch_tag, std::forward<_ExecutionPolicy>(__exec),
-                                                             std::forward<_R>(__r));
-        return std::ranges::borrowed_iterator_t<_R>{__last};
+                                                             __r);
+        return std::ranges::begin(__r) + std::ranges::size(__r);
```

Comment:
Would not the original approach be better? If the range is an r-value, it is moved (not copied), and the iterator type to return would have been `std::ranges::dangling`.